web/docs/database/operations/queries.md

```diff
@@ -0,0 +1,666 @@
+
+## Options Reference
+
+### Declaring Queries
+
+The `query` declaration supports the following fields:
+
+- `fn: ServerImport` <Required />
+
+  The import statement of the Query's NodeJs implementation.
+
+- `entities: [Entity]`
+
+  A list of entities you wish to use inside your Query.
+  For instructions on using Entities in Queries, take a look at [the guide](#using-entities-in-queries).
+
+#### Example
+
+<Tabs groupId="js-ts">
+<TabItem value="js" label="JavaScript">
+
+Declaring the Query:
+
+```wasp
+query getFoo {
+    fn: import { getFoo } from "@server/queries.js"
+    entities: [Foo]
+}
+```
+
+Makes the following import on both the server and the client:
+
+```js
+import getFoo from "@wasp/queries";
+```
+
+</TabItem>
+<TabItem value="ts" label="TypeScript">
+
+Declaring the Query:
+
+```wasp
+query getFoo {
+    fn: import { getFoo } from "@server/queries.js"
+    entities: [Foo]
+}
+```
+
+Makes the following imports available on both the server and the client:
+
+```ts
+import getFoo from "@wasp/queries";
+```
+
+As well as the following type import on the server:
+
+```ts
+import type { GetFoo } from "@wasp/queries/types";
+```
+
+</TabItem>
+</Tabs>
+
+### Implementing Queries
+
+The Query's implementation is a NodeJS function that takes two arguments (it can be an `async` function if you need to use the `await` keyword).
+Since both arguments are positional, you can name the parameters however you want, but we'll stick with `args` and `context`:
+
+1. `args` (type depends on the Query)
+
+   An object containing all the arguments (i.e., payload) **passed to the Query by the caller** (e.g., filtering conditions).
+   Check [the usage examples](#using-the-query) to see how to pass this object to the Query.
+
+2. `context` (type depends on the Query)
+
+   An additional context object **passed into the Query by Wasp**. This object contains user session information, as well as information about entities. Check the [section about using entities in Queries](#using-entities-in-queries) to see how to use the `context` object.
+
+<ShowForTs>
+
+Afer you [declare the query](#declaring-queries-in-wasp), Wasp generates a generic type you can use when defining its implementation.
+For the Query declared as `getSomething`, the generated type is called `GetSomething`:
+
+```ts
+import { GetSomething } from "@wasp/queries/types";
+```
+
+It expects two (optional) type arguments:
+
+1.  `Input`
+
+    The type of the `args` object (i.e., the Query's input payload). The default value is `never`.
+
+2.  `Output`
+
+    The type of the Query's return value (i.e., the Query's output payload). The default value is `unknown`.
+
+The defaults were chosen to make the type signature as permissive as possible. If don't want your Query to take/return anything, use `void` as a type argument.
+
+</ShowForTs>
+
+#### Example
```

Comment:
This example shows usage of the generic type, but it offers a Javascript impl where that doesn't make sense (there are no types used).

I would suggest wrapping the whole section, starting with "Afer you [declare the query](#declaring-queries-in-wasp), Wasp generates a generic type you can use when defining its implementation." in `<ShowForTs>` and only have the example for TS.

Reply:
That's exactly what I did, but forgot to import it :upside_down_face: